inc/switchtec/switchtec.h

```diff
@@ -405,6 +413,51 @@ enum switchtec_fw_active_flags {
 	SWITCHTEC_FW_PART_RUNNING = 2,
 };
 
+/**
+*@brief
+* Structure of firmware image(.pmc) file header.
+* All bytes are in little endian.
+*/
+typedef struct fw_image_header
+{
+	unsigned char vendor_id[4];	//!< Vendor ID, e.g. "MSCC" */
+	unsigned char img_length[4];	//!< image length = firmware code len + sign_len */
+	unsigned char part_type[4];	//!< Partition type, refers to flayout_part_type_enum */
+	unsigned char load_addr[4];	//!< Firmware load address (unused) */
+	unsigned char version[4];	//!< Firmware version */
+	unsigned char sequence[4];	//!< Sequence number (unused) */
+	unsigned char auth_en[4];	//!< Indicates the download file(.pmc) has signature data or not */
+	unsigned char sign_len[4];	//!< The length of the signature data */
+	unsigned char xml_version[4];	//!< Xml version supported */
+	unsigned char date_str[8];	//!< Format: YYYYMMDD, eg, 20171106 */
+	unsigned char time_str[8];	//!< Format: HH:MM:SS, 16:55:10 */
+	unsigned char img_str[16];	//!< Descriptions string for current image, eg, MAIN FW */
+	unsigned char bl_baud_rate;	//!< Bootloader baud rate, refer to uart_baud_enum */
+	unsigned char stdio_uart_port;	//!< UART port used in bootloader and main fw before reading cfg file */
+	unsigned char reserved[178];	//!< unused */
+	unsigned char hdr_crc[4];	//!< header CRC value for 256B header */
+	unsigned char img_crc[4];	//!< image CRC value for valid fimrware code/data without signature data */
+}__attribute__((packed)) fw_image_header;
+struct switchtec_fwdl_meta
```

Comment:
ditto

Reply:
i know the footer is replaced with metadata, my question is why it's not in fw.c but moved to switchtec.h?